### PR TITLE
[package] fix persistent_load error

### PR DESCRIPTION
**Pull request by lobstergrindset**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#58439 [package] fix persistent_load error**

Differential Revision: [D28494250](https://our.internmc.facebook.com/intern/diff/D28494250)

### CODE REVIEW

**Comment by Lilyjjo** [comment]
@Lilyjjo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28494250).

File: torch/package/package_importer.py [hunk 1]
```diff
@@ -208,6 +208,9 @@ def persistent_load(saved_id):
                 storage = loaded_storages[key]
                 return storage
             elif typename == "reduce_package":
+                if len(data) == 2:
```
Comment (suo) [inline torch/package/package_importer.py:211]: maybe add a comment that this is for BC, and causes produced objects not to be shared properly

**Review by suo — APPROVED** [review]

**Comment by Lilyjjo** [comment]
@Lilyjjo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28494250).

**Comment by Lilyjjo** [comment]
@Lilyjjo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28494250).

**Pull request closed without merging**